### ISSUE TITLE
problem with logger in 1.0.1 version

### ISSUE THREAD

**Issue by senid231** [rocketjob/net_tcp_client#6]
i use this `net_tcp_client` for my JSON RPC client [jrpc](https://github.com/senid231/jrpc)
but when i tried to use simple `Logger.new($stdout)` i faced with bug: block for `logger.benchmark_debug` in `read()` was called twice and return only result of last reading.
all works fine when i use github version, but i can't add github dependency to .gemspec

without logger all works fine.

should i wait new version release? or maybe i can fix this in a different way?

**Comment by senid231** [comment]
some backtrace:
```
D, [2015-12-02T10:11:42.052533 #14776] DEBUG -- :  #write ==> sending -- "107:{\"jsonrpc\":\"2.0\",\"method\":\"yeti.show\",\"params\":[\"system\",\"status\"],\"id\":\"UEc4HOeQWJY1LGHCmQGZIB1SlFoxqIrM\"},"
D, [2015-12-02T10:11:42.052659 #14776] DEBUG -- :  (0.0ms)  #write ==> complete -- 112
/home/senid/projects/my/jrpc/lib/jrpc/tcp_client.rb:43:in `receive_response': invalid response. missed comma as terminator (JRPC::ClientError)
	from /home/senid/projects/my/jrpc/lib/jrpc/tcp_client.rb:25:in `send_command'
	from /home/senid/projects/my/jrpc/lib/jrpc/base_client.rb:25:in `invoke_request'
	from /home/senid/projects/my/jrpc/cmd.rb:20:in `<top (required)>'
	from -e:1:in `load'
	from -e:1:in `<main>'
D, [2015-12-02T10:11:42.257174 #14776] DEBUG -- :  #read <== received -- "2"
D, [2015-12-02T10:11:42.257240 #14776] DEBUG -- :  #read <== received -- "4"
D, [2015-12-02T10:11:42.257281 #14776] DEBUG -- :  (204.5ms)  #read <== read 1 bytes
D, [2015-12-02T10:11:42.257316 #14776] DEBUG -- :  #read <== received -- "5"
D, [2015-12-02T10:11:42.257338 #14776] DEBUG -- :  #read <== received -- ":"
D, [2015-12-02T10:11:42.257355 #14776] DEBUG -- :  (0.0ms)  #read <== read 1 bytes
D, [2015-12-02T10:11:42.257379 #14776] DEBUG -- :  #read <== received -- "{\"id\""
D, [2015-12-02T10:11:42.257399 #14776] DEBUG -- :  #read <== received -- ": \"UE"
D, [2015-12-02T10:11:42.257415 #14776] DEBUG -- :  (0.0ms)  #read <== read 5 bytes
```

some code that invokes it:
```ruby
def receive_response
      length = get_msg_length
      response = @client.read(length+1)
      raise ClientError.new('invalid response. missed comma as terminator') if response[-1] != ','
      response.chomp(',')
    end

    def get_msg_length
      length = ''
      while true do
        character = @client.read(1)
        break if character == ':'
        length += character
      end

      Integer(length)
    end
```

```ruby
@client = Net::TCPClient.new server: @uri,
                                   connect_retry_count: t,
                                   connect_timeout: t,
                                   read_timeout: t,
                                   buffered: false,
                                   logger: @options.delete(:logger)
```


as you see it reading 2 bytes instead of one and return only last byte

**Comment by reidmorrison** [comment]
@tmgardner fixed this issue for us in master. He is currently working on adding SSL support at which point we were going to release a new gem.

Since you have another gem that needs this fix ASAP, I can create a new gem with the latest code from master?

**Comment by reidmorrison** [comment]
Published gem v1.0.2 to fix the double benchmark problem. Please let me know if there are any issues.

@senid231, we are busy with v2 in master, let me know if there are any other features you believe should be in `Net::TCPClient`

**Comment by senid231** [comment]
works fine :+1: thanks
